sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts

```diff
@@ -343,4 +385,139 @@ export class AssessmentSequentialProcessingService {
       options?.alternativeReferenceDate,
     );
   }
+
+  /**
+   * Get SFAS application awards totals.
+   * @param assessment current assessment being processed.
+   * @param firstAssessmentDate date of the first assessment date of the current application.
+   * @returns SFAS application awards totals.
+   */
+  async getProgramYearSFASAwardsTotals(
+    assessment: StudentAssessment,
+    firstAssessmentDate: string,
+  ): Promise<AwardTotal[]> {
+    const lastName = assessment.application.student.user.lastName;
+    const sin = assessment.application.student.sinValidation.sin;
+    const birthDate = assessment.application.student.birthDate;
+    const programYearStartDate = assessment.application.programYear.startDate;
+    const sfasApplicationAwards = this.sfasApplicationsRepo
+      .createQueryBuilder("sfasApplication")
+      .select("SUM(sfasApplication.csgp_award)", "CSGP")
+      .addSelect("SUM(sfasApplication.sbsd_award)", "SBSD")
+      .addSelect("SUM(sfasApplication.csgd_award)", "CSGD")
+      .addSelect("SUM(sfasApplication.bcag_award)", "BCAG")
+      .innerJoin("sfasApplication.individual", "sfasStudent")
+      .where("lower(sfasStudent.lastName) = lower(:lastName)", { lastName })
+      .andWhere("sfasStudent.sin = :sin", { sin })
+      .andWhere("sfasStudent.birthDate = :birthDate", { birthDate })
+      .andWhere("sfasApplication.startDate >= :startDate", {
+        startDate: programYearStartDate,
+      })
+      .andWhere("sfasApplication.endDate < :endDate", {
+        endDate: firstAssessmentDate,
+      });
+
+    const awards =
+      await sfasApplicationAwards.getRawOne<SFASApplicationAwards>();
+    const totals: AwardTotal[] = [];
+    if (awards) {
+      Object.entries(awards).forEach(([key, value]) => {
+        if (+value > 0) {
+          totals.push({
+            offeringIntensity: OfferingIntensity.fullTime,
+            valueCode: key,
+            total: +value,
+          });
+        }
+      });
+    }
```

Comment:
As a suggestion, we use `map` instead of `for` under the understanding that it looks better for readability.